### ISSUE TITLE
Wrongly generated URL when hashtag in index name 

### ISSUE THREAD

**Issue by rayrutjes** [algolia/algoliasearch-client-go#574]
* Algolia for Go (3.5.0);
* Go (go1.13.4)

### Description
It seems like additional parameters like `getVersion=2` get interleaved with the index name when it contains a hashtag.
I suppose the hashtag needs to be encoded so that it is not considered as an actual hashtag when leveraging URL helpers.

### How ro reproduce
```go
package main

import (
	"log"

	"github.com/algolia/algoliasearch-client-go/v3/algolia/debug"
	"github.com/algolia/algoliasearch-client-go/v3/algolia/search"
)

func main() {
	debug.Enable()
	c := search.NewClient("YYY", "XXX")
	i := c.InitIndex("indexname@algolia.com#whatever")
	s, err := i.GetSettings()
	if err != nil {
		log.Printf("err: %v", err)
		return
	}

	log.Printf("settings: %v", s)
}

```

outputs:

```
ALGOLIA DEBUG: > ALGOLIA DEBUG request:
	method="GET"
	url="https://YYY-dsn.algolia.net/1/indexes/indexname@algolia.com?getVersion=2#whatever/settings"
	header=User-Agent:"Algolia for Go (3.5.0);Go (go1.13.4)"
	header=X-Algolia-Application-Id:"**********"
	header=X-Algolia-Api-Key:"********************************"
	header=Connection:"Keep-Alive"
	header=Content-Type:"application/json; charset=utf-8"
	body=
```

### How this was discovered
Using an API key with restriction strictly on the full index name made the above request fail because it would target `indexname@algolia.com` instead of `indexname@algolia.com#whatever`.

**Comment by aseure** [comment]
@rayrutjes This should finally be fixed in v3.6.0 that we will release hopefully Monday, March 2nd 2020. Thanks for your patience on this one.

**Comment by aseure** [comment]
@rayrutjes We had to delay the release because of the warning email we wanted to send to our customers which took us more time than expected. We've just released [v3.6.0](https://github.com/algolia/algoliasearch-client-go/blob/master/ChangeLog.md#v360-2020-03-09) with the fix. Thanks again for reporting!
